cub/cub/device/device_segmented_reduce.cuh

```diff
@@ -73,6 +78,66 @@ CUB_NAMESPACE_BEGIN
 //! @endrst
 struct DeviceSegmentedReduce
 {
+private:
+  template <typename InputIteratorT,
+            typename OutputIteratorT,
+            typename BeginOffsetIteratorT,
+            typename EndOffsetIteratorT,
+            typename OffsetT,
+            typename ReductionOpT,
+            typename InitT>
+  CUB_RUNTIME_FUNCTION static cudaError_t custom_segmented_reduce(
```

Comment:
suggestion: I believe `custom_` was used in the radix sort code to denote custom types support (user-defined). This is not related to the check we have in segmented sort. I'd go with something like:
```suggestion
  CUB_RUNTIME_FUNCTION static cudaError_t segmented_reduce(
```